src/codegen.rs

```diff
@@ -83,9 +86,15 @@ pub struct CodeGen<'ink> {
 
 pub struct GeneratedModule<'ink> {
     module: Module<'ink>,
+    location: PathBuf,
     engine: RefCell<Option<ExecutionEngine<'ink>>>,
 }
 
+/// We need to be able to process the modules within threads
+/// TODO: is this really correct, is an LLVM module thread safe?
+unsafe impl<'ink> Send for GeneratedModule<'ink> {}
+unsafe impl<'ink> Sync for GeneratedModule<'ink> {}
```

Comment:
I don't get any errors when removing these lines, meaning they're not neccessary after all? Also doesn't the `Sync` seem wrong here given we have a `RefCell` which is not thread-safe?

Reply:
You're right, these are left over from previous prototyping approaches. Good catch